src/commonMain/kotlin/io/github/detekt/sarif4k/Merging.kt

```diff
@@ -0,0 +1,50 @@
+@file:JvmName("Merging")
+package io.github.detekt.sarif4k
+
+import kotlin.jvm.JvmName
+
+fun SarifSchema210.merge(other: SarifSchema210): SarifSchema210 {
+    require(schema == other.schema) { "Cannot merge sarifs with different schemas: '$schema' or '${other.schema}'" }
+    require(version == other.version) { "Cannot merge sarifs with different versions: '$version' or '${other.version}'" }
+
+    val mergedExternalProperties = (inlineExternalProperties.orEmpty() + other.inlineExternalProperties.orEmpty())
+        .takeIf { it.isNotEmpty() }
+
+    val mergedProperties = properties.merge(other.properties)
+
+    val mergedRuns = runs.merge(other.runs)
+
+    return SarifSchema210(
+        schema,
+        version,
+        mergedExternalProperties,
+        mergedProperties,
+        mergedRuns
+    )
+}
+
+fun PropertyBag?.merge(other: PropertyBag?): PropertyBag? {
+    return (this?.tags.orEmpty() + other?.tags.orEmpty())
+        .takeIf { it.isNotEmpty() }
+        ?.let { PropertyBag(it) }
+}
+
+fun List<Run>.merge(other: List<Run>): List<Run> {
+    val runsByTool = (this + other).groupBy { it.tool.driver.fullName }
```

Comment:
> Thanks! Probably, in the future, we will need more tests but these is just a port from detekt so no need to add them in this PR.

@BraisGabin I think this is not just the port of the merger from Detekt, but also extra logic from https://github.com/detekt/detekt/pull/6894, with an extra revision on top of it.

Reply:
Yes, but that part I think it have tests. So, to me we can merge and prepare a release. Do you know how to do that?

Reply:
@BraisGabin https://github.com/detekt/sarif4k/blob/main/RELEASING.md